### DISCUSSION TITLE
How to create format item from Japanese datetime string?

### DISCUSSION THREAD

**Discussion by thelastfantasy** [time-rs/time#557] [Question]
Just like this
```rust
// This is a used commonly used schedule format in Japanese TV shows
// Equals to "03/09(Thusday) 25:00" and "03/10(Friday) 01:00"
// Other examples can be seen here: https://www.kansou.me/
let time_str = "03/09(木) 25:00"; 
let format = format_description::parse("[month]/[day]([weekday]) [hour]:[minute]").unwrap();

let input_time = OffsetDateTime::parse(time_str, &format).unwrap();

println!(
    "{}",
    input_time.to_offset(offset!(+9)).format(&format).unwrap()
);
```

I have 2 problems in this code:

1. How to parse Japanese weekdays from Japanese datestrings just like the full weekday string "木曜日" or the short "木"
2. How to parse hour over than 24

This is the full Japanese weekday list from Sunday to Saturday.
Full format: "日曜日", "月曜日", "火曜日", "水曜日", "木曜日", "金曜日", "土曜日"
The short:  "日", "月", "火", "水", "木", "金", "土"

**Comment by jhpratt** [comment]
While I understand that Japan does this (and a few other countries I believe), this is simply not possible to handle. Days only have 24 hours in them, so anything beyond that must be rejected as invalid. Even without that, there is no year present in the formatted value, so the date could not be parsed.

**Comment by golddranks** [comment]
@thelastfantasy I have written some data mangling and visualization code for Japanese TV show data. The simplest sensible thing I found to work is to write a parser of the time part myself, and convert that to seconds from start of the date indicated by the date part. Then you can add the seconds to get a datetime, or whatever. In my case, it was easier just to keep stuff in seconds, and use datetimes objects as "date anchors" the seconds were relative to. Then, implement a display routine for the time part from seconds to string for output.
